src/UndoRedo.spec.ts

```diff
@@ -2,4 +2,130 @@
 declare var require: NodeRequire;
 
 import test = require('tape');
-import undoRedo from './';
+import undoRedoStack from './';
+
+interface countObj {
+    undoCount: number;
+    redoCount: number;
+}
+
+function createObject(object: countObj) {
+    const obj = {
+        undo: () => {
+            object.undoCount++;
+            return true;
+        },
+        redo: () => {
+            object.redoCount++;
+            return true;
+        }
+    };
+    return obj;
+}
+
+function undo(undoRedo: UndoRedo, numTimes: number = 1) {
+    for (let i = 0; i < numTimes; i++) {
+        undoRedo.doUndo();
+    }
+}
+
+function redo(undoRedo: UndoRedo, numTimes: number = 1) {
+    for (let i = 0; i < numTimes; i++) {
+        undoRedo.doRedo();
+    }
+}
+
+function addUndo(undoRedo: UndoRedo, obj: Object, numTimes: number = 1) {
+    for (let i = 0; i < numTimes; i++) {
+        undoRedo.addUndoObject(obj);
+    }
+}
+
+test(`should call both doUndo() and doRedo() 1 time`, t => {
+    const ur = undoRedoStack.clear();
```

Comment:
i think what you mean is just
`undoRedoStack.clear();`
And rather than using `ur` just use `undoRedoStack` everywhere.